roles/docker/tasks/main.yml

```diff
@@ -113,4 +103,14 @@
   notify:
     - restart docker
```

Comment:
Not the very first time round, because the task `Start the Docker service` starts docker, registers the result in `start_result` and then we have:

```
- set_fact:
    docker_service_status_changed: start_result | changed
```

And the `restart docker` handler only runs

```
  when: not docker_service_status_changed | default(false) | bool
```